### PR TITLE
[APB-4804][JR] Implement Basic Auth for terminations

### PR DESCRIPTION
**Pull request by jordanrowe**

### CODE REVIEW

File: app/uk/gov/hmrc/agentfirelationship/connectors/AgentClientAuthConnector.scala [hunk 1]
```diff
@@ -73,6 +78,38 @@ class AgentClientAuthConnector @Inject()(val authConnector: AuthConnector)(impli
...
+              Logger.warn(
+                "Authorization header found in the request but invalid username or password")
+              Future successful Unauthorized
+            }
+          case _ =>
+            Logger.warn(
+              "Authorization header found in the request but its not in the expected format")
+            Future successful Unauthorized
+        }
+      case _ =>
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentfirelationship/connectors/AgentClientAuthConnector.scala]: i think unit tests are missing for all these edge cases ?
